File: EstructurasDeDatosII/PythonNayelli/Ejercicios/ALGORITMOS_GREEDY/monedas.cpp
#include <bits/stdc++.h>
using namespace std;

int cant, cien, cincuenta, veinte, diez, cinco, dos, uno;

void cambio(int cant, int &cien, int &cincuenta, int &veinte, int &diez, int &cinco, int &dos, int &uno){
    cien+=cant/100; cant%=100;
    cincuenta+=cant/50; cant%=50;
    veinte+=cant/20; cant%=20;
    diez+=cant/10; cant%=10;
    cinco+=cant/5; cant%=5;
    dos+=cant/2; cant%=2;
    uno+=cant;
    return;
}

int main(){

    cout<<"Ingrese una cantidad: "; cin>>cant;
    cambio(cant, cien, cincuenta, veinte, diez, cinco, dos, uno);
    cout<<"Cantidad en billetes/monedas.\n";
    cout<<"100$: "<<cien<<'\n';
    cout<<"50$: "<<cincuenta<<'\n';
    cout<<"20$: "<<veinte<<'\n';
    cout<<"10$: "<<diez<<'\n';
    cout<<"5$: "<<cinco<<'\n';
    cout<<"2$: "<<dos<<'\n';
    cout<<"1$: "<<uno<<'\n';
}

File: EstructurasDeDatosII/PythonNayelli/Ejercicios/ALGORITMOS_GREEDY/monedas.py

cant, v = 0, [0,0,0,0,0,0,0]

def cambio(cant,v):
    v[0]+=cant//100
    cant%=100
    v[1]+=cant//50
    cant%=50
    v[2]+=cant//20
    cant%=20
    v[3]+=cant//10
    cant%=10
    v[4]+=cant//5 
    cant%=5
    v[5]+=cant//2
    cant%=2
    v[6]+=cant
    cant+=1
    return

cant = int(input("Ingrese una cantidad: "))
cambio(cant,v)

print("Cantidad en billetes/monedas.")
print("100$: "+str(v[0]))
print("50$: "+str(v[1]))
print("20$: "+str(v[2]))
print("10$: "+str(v[3]))
print("5$: "+str(v[4]))
print("2$: "+str(v[5]))
print("1$: "+str(v[6]))


